src/openforms/registrations/contrib/zgw_apis/plugin.py

```diff
@@ -401,6 +437,16 @@ def register_submission(self, submission: Submission, options: dict) -> dict | N
         if submission.has_registrator:
             result["medewerker_rol"] = medewerker_rol
 
+        # Register submission to Objects API if configured
+        if (
+            options.get("objecttype")
+            and options.get("objecttype_version")
+            and options.get("content_json")
+        ):
+            result = self.register_submission_to_objects_api(
+                submission, options, result
+            )
```

Comment:
I find the API here a bit unusual, as `register_submission_to_objects_api` mutates `result` and returns it. I think it would be better to return the object API result and set `result["objects_api"] = res` after the call to the method. Might be related to my other comment

Reply:
Yes, you are right!It 's not clear enough the way I did it, I will fix that thanks.